src/market/MarketPlaceManagerActions.cc

```diff
@@ -69,35 +69,35 @@ int MarketPlaceManager::import_app(
     switch (type)
     {
         case MarketPlaceApp::IMAGE:
+        {
+            int ds_id;
+
+            if ( auto image = ipool->get_ro(origin_id) )
             {
-                int ds_id;
-
-                if ( auto image = ipool->get_ro(origin_id) )
-                {
-                    image->to_xml(image_data);
-
-                    ds_id = image->get_ds_id();
-                }
-                else
-                {
-                    goto error_noimage;
-                }
-
-                if ( auto ds = dspool->get_ro(ds_id) )
-                {
-                    ds->to_xml(ds_data);
-                }
-                else
-                {
-                    goto error_nods;
-                }
-
-                if (imagem->set_app_clone_state(app_id, origin_id, err) != 0)
-                {
-                    goto error_clone;
-                }
+                image->to_xml(image_data);
+
+                ds_id = image->get_ds_id();
             }
-            break;
+            else
+            {
+                goto error_noimage;
+            }
+
+            if ( auto ds = dspool->get_ro(ds_id) )
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This snippet may contain an accidental use of incorrect operation. Make sure that the operation being performed is in fact what you intended.